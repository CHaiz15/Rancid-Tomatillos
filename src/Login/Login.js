import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import { fetchUser, getRatings } from '../apiCalls';
import { setUser, setRatings } from '../Actions';
import { connect } from 'react-redux';
import './Login.scss';

export class Login extends Component {
  constructor() {
    super();
    this.state = { 
      email: '',
      password: 654321
    }
  }

  handleSubmit = () => {  
    const { email, password } = this.state;
    fetchUser( email, password ) 
      .then(data => {
        getRatings(data.user.id)
          .then(ratings => {
            this.props.setUser(data)
            this.props.setRatings(ratings)
          })
      })
  }

  handleChange = (e) => {
    this.setState({[e.target.name]: e.target.value})
  }

  render() {
    return(
      <section className='login-container'>
        <form className='login-form'> 
          <label>Email:</label>
          <input 
            className='email-input'
            name='email' 
            value={this.state.email} 
            onChange={(e) => this.handleChange(e)}
          />
          <label>Password:</label>
          <input 
            className='password-input'
            type='password'
            name='password' 
            value={this.state.password} 
            onChange={(e) => this.handleChange(e)}
          />
          <NavLink
            className='login_button'
            to='/'
            type='button'
            onClick={this.handleSubmit}
          >ENTER </NavLink>
        </form>
      </section>
    )
  }
}

export const mapStateToProps = (state) => ({
  user: state.user
})

export const mapDispatchToProps = dispatch => ({
  setUser: user => dispatch(setUser(user)),
  setRatings: ratings => dispatch(setRatings(ratings))
})

export default connect(mapStateToProps, mapDispatchToProps)(Login);
