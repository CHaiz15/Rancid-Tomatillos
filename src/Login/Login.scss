.login-container {
  margin-top: 5rem;
  display: flex;
  align-items: center;
  flex-direction: column;
  flex-wrap: wrap;
  height: 41.2rem;
  background: linear-gradient(rgba(0, 0, 0, 0.3), rgba(0, 0, 0, 0.3)), url('../assets/film-bg.jpg');
  background-position: bottom, left;
}

.login-form {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-around;
  background-color: #fff;
  margin: 5rem 0 0 0;
  height: 18rem;
  width: 50%;
}

.password-input, .email-input {
  font-size: 15px;
  height: 30px;
  margin: 2% 0 0 0;
  width: 80%;
}

.login_button {
  padding: 1%;
  border-radius: 30px;
  font-size: 18px;
  width: 7rem;
  text-align: center;
  background-color: #A16447;
  color: white;
  font-weight: 600;
  box-shadow: 0 3px #999;
}

.login_button:hover {
  background-color: rgb(107, 57, 34);
}

