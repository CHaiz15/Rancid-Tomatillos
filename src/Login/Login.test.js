import React from 'react';
import { Login, mapStateToProps, mapDispatchToProps } from './Login';
import { shallow } from 'enzyme';

describe('Login', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<Login />);
    expect(wrapper).toMatchSnapshot();
  });
});