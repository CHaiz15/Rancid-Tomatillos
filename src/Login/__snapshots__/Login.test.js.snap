// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should match the snapshot 1`] = `
<section
  className="login-container"
>
  <form
    className="login-form"
  >
    <label>
      Email:
    </label>
    <input
      className="email-input"
      name="email"
      onChange={[Function]}
      value=""
    />
    <label>
      Password:
    </label>
    <input
      className="password-input"
      name="password"
      onChange={[Function]}
      type="password"
      value={654321}
    />
    <NavLink
      className="login_button"
      onClick={[Function]}
      to="/"
      type="button"
    >
      ENTER 
    </NavLink>
  </form>
</section>
`;
